File: backend/src/main/java/com/apurv/environment/dto/EnvironmentResponse.java
package com.apurv.environment.dto;

import java.util.List;
import java.util.ArrayList;
import java.util.UUID;

import com.apurv.environment.entity.EnvironmentColor;

import java.time.Instant;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EnvironmentResponse {

    private UUID id;
    private UUID collectionId;
    private String name;
    private EnvironmentColor environmentColor;
    @Builder.Default
    private List<VariableResponse> variables = new ArrayList<>();
    private Instant createdAt;
    private Instant updatedAt;

}

File: frontend/src/environment/types/response/EnvironmentResponse.ts
import type VariableResponse from "./VariableResponse";
import type { EnvironmentColor } from "../items/Environment";

export default interface EnvironmentResponse {
  id: string;
  name: string;
  environmentColor: EnvironmentColor;
  variables: VariableResponse[];
  createdAt: string;
  updatedAt: string;
}
